### ISSUE TITLE
Breadcrumb is displaying wrong "path" order when collection in moved inside another

### ISSUE THREAD

**Issue by rodolphogarrido** [metabase/metabase#13053]
**Describe the bug**
Breadcrumb is displaying wrong "path" order when collection in moved inside another.

**To Reproduce**
1. Create a new collection named "Collection**1**" in "Our analytics".

2. Create another collection named "Collection**2**" with a collection named "Collection**3**" inside it, also in "Our analytics", so when we enter "Collection**3**" its breadcrumb is :
"Our analytics/Collection**2**/Collection**3**"

3. Move "Collection**2**" inside "Collection**1**", so when we enter "Collection**3**" its breadcrumb is :
"Our analytics/Collection**1**/Collection**2**/Collection**3**"

4. Create another collection named "Collection**4**".

5. Move "Collection**1**" inside "Collection**4**".

6. Enter "Collection**3**" and see its breadcrumb shown as:
"Our analytics/Collection**1**/Collection**2**"/Collection**3**/Collection**4**"

**Expected behavior**
Breadcrumb display order should be:
"Our analytics/Collection**4**/Collection**1**/Collection**2**/Collection**3**/"
Note in the video "Collection**4**" appears last.

**Screenshots**
![breadcrumb-issue](https://user-images.githubusercontent.com/47122489/89080778-394ead00-d360-11ea-9167-1f4ab63df467.gif)

**Severity**
It is annoying some users.

**Additional context**
N/A

**Metabase Diagnostic Info**

```json
{
  "browser-info": {
    "language": "pt-BR",
    "platform": "Linux x86_64",
    "userAgent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4147.105 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.5+10",
    "java.vendor": "AdoptOpenJDK",
    "java.vendor.url": "https://adoptopenjdk.net/",
    "java.version": "11.0.5",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.5+10",
    "os.name": "Linux",
    "os.version": "4.19.106-coreos",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "postgres",
      "sqlserver",
      "druid",
      "googleanalytics",
      "presto",
      "h2"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "11.5"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.2.8"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2020-02-25",
      "tag": "v0.34.3",
      "branch": "release-0.34.x",
      "hash": "1a83edb"
    },
    "settings": {
      "report-timezone": "UTC"
    }
  }
}
```

**Comment by flamber** [comment]
Hi @rodolphogarrido 
Yeah, it seems like it's sorting by name. I'm ending up with the same mix on 0.36.2:
![image](https://user-images.githubusercontent.com/1447303/89100722-a3fff700-d3f9-11ea-96bd-aac1c0396da8.png)

**Comment by nemanjaglumac** [comment]
`location` info displays correct order. Here's a screenshot from a React Dev Tools:

![image](https://user-images.githubusercontent.com/31325167/89113137-473e2400-d46d-11ea-8a71-aa8952488b0f.png)

`location` is "converted" to `id`. If we follow the location path (6/3/10/4) it would give us the correct order. In my case, it was:
Collection 4 > Collection 1 > Collection 5 > Collection 2.
Note that `parent_id` also points to the correct item.

- BreadCrumbs component is simply mapping over `ancestors` to produce breadcrumbs.
- `effective_ancestors` is already sorted and that seems to be the reason for this behaviour

#### Source:
`src/metabase/models/collection.clj` lines 315-316:
``` clojure
...
(when-let [ancestor-ids (seq (location-path->ids location))]
    (db/select [Collection :name :id] :id [:in ancestor-ids] {:order-by [:%lower.name]})))
```

Hope this helps.

**Comment by nemanjaglumac** [comment]
So, I've been playing with this locally. Managed to solve it, but it feels overly complicated. There is probably some "low level" solution/fix that I couldn't find.  

This is literary just me thinking out loud and testing my understanding of the code base:

#### Backend
- we get `effective_ancestors` from SQL command (`db/select ...`)
    - even if we remove `{order-by [:%lower.name]}` SQL still applies some default sort
    - I couldn't find any other parameter to sort by that would keep the order we get from `ancestor-ids` seq
- next thought was to somehow loop/map through each individial item in a seq
    - but that would require separate call to the DB for each iteration (we probably want to keep those to a minimum?)
    - instead of `WHERE id IN (x, y, z)` we'd use something like `WHERE id=x`

Was it right to assume that it would be "safer" (and probably cheaper - performance wise) to use rearranged instance of the array in appropriate React component `frontend/src/metabase/components/CollectionLanding.jsx`?

#### Frontend
- as we've seen, `effective_ancestors` already comes sorted by name from the backend
- `collection.location` keeps the correct order -> it can serve as a blueprint for the new array
    - we need to first convert it from a string into array of integers `"/6/3/10/4/" => [6, 3, 10, 4]`
- we can now filter `ancestors` array (testing if `ancestor.id === id`) into a newly created `sortedAncestors` array and use it to create crumbs

I don't like that this solution is O(n²) but so far I couldn't think of anything better.

**Comment by rodolphogarrido** [comment]
Thanks for the concern and for the fast reply. :smile:

**Comment by nemanjaglumac** [comment]
We have re-imagined collections and vastly improved the user experience in the newest [release 0.38.0](https://github.com/metabase/metabase/releases/tag/v0.38.0). Have you had a chance to try it out @rodolphogarrido ?

**Comment by flamber** [comment]
We don't show a breadcrumb path anymore, since there's a hierarchy in the sidebar, so I think we can close this.

**Comment by rodolphogarrido** [comment]
Hi @nemanjaglumac thanks for the heads up, I will try it out soon 😄

**Comment by lajkomiki** [comment]
Hi All, 

With the breadcrumbs being back in the newer versions (1.43/1.44 ?) this seems to be an issue again. 
I attach some screenshots. Except for the deepest collection, the rest is displayed according to an alphabetic order. 

<img width="1728" alt="incorrectly displayed collection path in breadcrumb" src="https://user-images.githubusercontent.com/34609670/185635140-a53a4b75-e182-4885-9dc7-f859d4cae8f7.png">
<img width="1688" alt="actual collection structure" src="https://user-images.githubusercontent.com/34609670/185635150-291760ab-c9bd-4348-a85b-97c9cab36f9f.png">

**Comment by iethree** [comment]
Could not reproduce on 48.1 - tried both postrges and h2 app dbs

possibly fixed by #23569 or #23727 or https://github.com/metabase/metabase/issues/22826



![Image](https://github.com/metabase/metabase/assets/30528226/abdee957-f366-4570-8fcf-fecc893633ab) ![Image](https://github.com/metabase/metabase/assets/30528226/44d3f817-99f5-4293-9fda-a49a75d6f1ed)




 ![Image](https://github.com/metabase/metabase/assets/30528226/0369987a-ce59-4d06-9911-83d7e3c2f9f2) ![Image](https://github.com/metabase/metabase/assets/30528226/e7de1ff7-193d-41b3-91bb-6d7b800616af)
